openapi.yaml

```diff
@@ -23,901 +31,1078 @@ tags:
+          $ref: '#/responses/401'
         '403':
-          description: 'This is used to indicate that the server can be reached and process the request but refuses to take any further action i.e. ''forbidden''. This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/403'
+        '404':
+          $ref: '#/responses/404'
         '422':
-          description: 'This error condition may occur if a JSON request body contains well-formed (i.e. syntactically correct), but semantically erroneous, JSON instructions. This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/422'
         '429':
-          description: 'The server is receiving too many requests i.e. ''server_busy''. Retry at a later time. This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/429'
         '500':
-          description: 'This code should be used only if there is catastrophic error and there is not a more appropriate code i.e. ''internal_server_error''. This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/500'
         default:
-          description: 'This is the default error response when no other appropriate code is available.  This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/default'
+parameters:
+  limit:
+    name: page[limit]
+    in: query
+    description: To define the download segmentation value i.e. the maximum number of records to be
+      contained in the response.
+    required: false
+    type: integer
+    format: int32
+    default: 100
+    minimum: 1
+    allowEmptyValue: false
+  offset:
+    name: page[offset]
+    in: query
+    description: The number of the first record to be supplied in the segmented response message.
+    required: false
+    type: integer
+    format: int32
+    default: 0
+    minimum: 0
+    allowEmptyValue: false
+  sort:
+    name: sort
+    in: query
+    description: "Identifies the sort criteria to be used for the records in the response message.
+    Order is ascending unless prefixed with a '-'"
+    required: false
+    type: string
+    allowEmptyValue: false
+  filter:
+    name: filter
+    in: query
+    description: 'The filtering rules to be applied when identifying the records to be supplied in
+    the response message. The sort order should follow the [UNICODE, 16] standard.'
+    required: false
+    type: string
```

Comment:
since the intention is to allow multiple filters, should this type be array with item type string, collectionFormat csv?

Reply:
Ah right, good catch. For our use-case, instead of having one big `filter` parameter, we would have multiple `filter[name][operation]` parameters so we end up deciding which filter options we support. Since we don't know which operations we'll need to support yet, I just kind of left this here as a placeholder.